analysis/src/code.rs

```diff
@@ -510,4 +551,17 @@ mod test {
             unsafe_details.functions + unsafe_details.functions
         );
     }
+
+    #[test]
+    fn test_code_exclusive_deps() {
+        let graph = get_test_graph();
+        let code_analyzer = get_test_code_analyzer();
+        let package = graph.packages().find(|p| p.name() == "gitlab").unwrap();
+        let dependencies = code_analyzer
+            .get_package_dependencies(&graph, &package)
+            .unwrap();
+        let exclusive_deps = code_analyzer.filter_exclusive_deps(&package, &dependencies);
+        // Checked by hand that it returns the right count :)
```

Comment:
This isn't a very good test if it requires us to check by hand ;) If you've verified the number then include that hardcoded here so that we can assert on it.

Reply:
If in future, we add packages in the test crate the hardcoded number would become invalid. How to handle that?

Reply:
One way would be to check in the output of a `cargo metadata` command and use it as a fixture just for this test. I believe the guppy crate does this for its own testing as well.